pydantic/_internal/_generate_schema.py

```diff
@@ -866,6 +870,11 @@ def _get_args_resolving_forward_refs(self, obj: Any) -> tuple[Any, ...] | None:
     def _get_args_resolving_forward_refs(self, obj: Any, required: bool = False) -> tuple[Any, ...] | None:
         args = get_args(obj)
         if args:
+            if not _typing_extra.is_literal_type(obj) and not _typing_extra.is_annotated(obj):
+                # This is a rare case, but one example is `obj = list['int']` in Python 3.9-3.11
+                # Other examples are `tuple['int'], type['int']`, etc.
+                # These cases are due to inconsistency between for example `list` and `List` in these versions.
+                args = [_typing_extra._make_forward_ref(a) if isinstance(a, str) else a for a in args]
```

Comment:
```suggestion
            if isinstance(obj, GenericAlias):
                # PEP 585 generic aliases don't convert args to ForwardRefs, unlike `typing.List/Dict` etc.
                args = (_typing_extra._make_forward_ref(a) if isinstance(a, str) else a for a in args)
```

Let's use the same check as in https://github.com/python/cpython/pull/30900.